src/qt/optionsmodel.cpp

```diff
@@ -244,10 +244,10 @@ void OptionsModel::SetPruneEnabled(bool prune, bool force)
     const int64_t prune_target_mib = PruneGBtoMiB(settings.value("nPruneSize").toInt());
     std::string prune_val = prune ? ToString(prune_target_mib) : "0";
     if (force) {
-        m_node.forceSetArg("-prune", prune_val);
+        gArgs.ForceSetArg("-prune", prune_val);
         return;
     }
-    if (!m_node.softSetArg("-prune", prune_val)) {
+    if (!gArgs.SoftSetArg("-prune", prune_val)) {
```

Comment:
Can't this use `this->node().context()->args` instead of the global, which should be gone hopefully :soon: ?

Reply:
re: https://github.com/bitcoin-core/gui/pull/35#discussion_r456261621

> Can't this use `this->node().context()->args` instead of the global, which should be gone hopefully 🔜 ?

After https://github.com/bitcoin/bitcoin/pull/10102, there is no NodeContext in the gui process. An ArgsManager reference could be passed to OptionsModel if desired, though.